pkg/iac/scanners/cloudformation/parser/parameter.go

```diff
@@ -27,7 +27,24 @@ func (p *Parameter) UnmarshalYAML(node *yaml.Node) error {
 }
 
 func (p *Parameter) UnmarshalJSONWithMetadata(node jfather.Node) error {
-	return node.Decode(&p.inner)
+
+	var inner parameterInner
+
+	if err := node.Decode(&inner); err != nil {
+		return err
+	}
+
+	// jfather parses Number without fraction as int64
+	// https://github.com/liamg/jfather/blob/4ef05d70c05af167226d3333a4ec7d8ac3c9c281/parse_number.go#L33-L42
+	switch v := inner.Default.(type) {
+	case int64:
+		inner.Default = int(v)
+	default:
+		inner.Default = v
+	}
+
+	p.inner = inner
+	return nil
```

Comment:
Should we extract this into a function? Looks like there are other callsites like this one: https://github.com/aquasecurity/trivy/blob/6a2f6fde4f97f254eb4ef3b79cab99f574abf72a/pkg/iac/scanners/cloudformation/parser/resource.go#L65

Reply:
The resource does not store a primitive value. This conversion is needed for the parameter and property of the resource.